src/main/java/net/snowflake/ingest/streaming/internal/RegisterService.java

```diff
@@ -195,8 +196,9 @@ List<FlushService.BlobData<T>> registerBlobs(Map<String, Timer.Context> latencyT
             Timer.Context registerContext =
                 Utils.createTimerContext(this.owningClient.registerLatency);
 
+            int blobsPerReq = getBlobNumToRegisterInOneReq(blobs.size());
             // Register the blobs, and invalidate any channels that return a failure status code
-            this.owningClient.registerBlobs(blobs);
+            Lists.partition(blobs, blobsPerReq).forEach(owningClient::registerBlobs);
```

Comment:
Since this is a IO networking request, would it make sense to parallelize?

Reply:
I would be generally in favour if it was a common case.
Here it should be more like exception indicating that system already overloaded (added a log),
Hence I tend to not complicate the system with even more concurrency for this.
Happy to discuss more

Reply:
Makes sense :)

Reply:
> I would be generally in favour if it was a common case.

Actually, I take this back, we cannot do this because of ordering guarantees per channel.

Reply:
@sfc-gh-azagrebin then you will need to use `forEachOrdered` as `forEach` has no ordering guarantees.